src/lib.rs

```diff
@@ -294,9 +294,7 @@ where
     T: crate::TimeSource,
 {
     fn drop(&mut self) {
-        self.volume_mgr
-            .close_volume(self.raw_volume)
-            .expect("Failed to close volume");
+        _ = self.volume_mgr.close_volume(self.raw_volume)
```

Comment:
I’m not sure about silently hiding these errors on drop. I think I would prefer to offer a non-panicking API and leave Drop as panicking. But I’m happy to be persuaded otherwise.

Reply:
I guess the non-panicking drop would just be to turn the `File` into a `RawFile` and close using the volume manager, or make a dedicated method for that on the `File` itself. My problem is just that if some part of a program relies on the `Drop` to close the file, then if the card is ejected or unresponsive, that will cause a panic. That to me is quite scary, if lets say, I am implementing a system where the logging module is very much *not* mission critical, but it still causes a crash on a bad connection.

I think the documentation for the `drop(..)` method explains the reason for avoiding panics being a general recommendation pretty well. But I also think think it is important to consider what behavior is desired/expected for the library.

Does an error during a close cause the program to be in an undefined or unrecoverable state? If not, I'd say the best-effort close is better than a panic. Perhaps it is only really relevant to consider the implications of the `File` drop, since causing the `Volume` or `Directory` drop to panic is not really possible as far as I can tell.

I just found it head-scratching that the program panicked when I first tried to test yanking the card during operation. I expected some data to not be flushed (before my previous commit), but not a panic. At the very least it should be loud and clear in the documentation for the `File` type that they will panic if they are not able to drop. Especially since actually encountering the panic during testing is not guaranteed if the card disconnecting is a very rate event.

Reply:
I was thinking that "Silent failures are bad", and drop can't do anything but panic on error. But std::fs::File says:

> Errors detected on closing are ignored by the implementation of Drop.

Reply:
Comparing to something similar in std library makes sense! And I think it goes with the idea of "do what the user would expect". Should we do it for File, Volume and Directory?

Reply:
Being consistent sounds good, if the change is well documented.